web3/providers/eth_tester/main.py

```diff
@@ -106,11 +106,11 @@ def make_request(self, method: RPCEndpoint, params: Any) -> RPCResponse:
                 "error": "RPC Endpoint has not been implemented: {0}".format(method),
             })
         except TransactionFailed as e:
-            if type(e.args[0]) == str:
-                reason = e.args[0]
-            else:
+            try:
                 reason = decode_single('(string)', e.args[0].args[0][4:])[0]
-            raise SolidityError(f'execution reverted: {reason}')
+            except (InsufficientDataBytes, AttributeError):
+                reason = e.args[0]
+            raise TransactionFailed(f'execution reverted: {reason}')
```

Comment:
2 of our test cases pick this up, but this can be fixed

I wish there were an ERC for revert reasons and how to handle them in web3 libraries